eotsmanager/store/storedsigningrecord.go

```diff
@@ -1,17 +1,29 @@
 package store
 
-import "github.com/babylonlabs-io/finality-provider/eotsmanager/proto"
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/babylonlabs-io/finality-provider/eotsmanager/proto"
+)
 
 type SigningRecord struct {
-	BlockHash []byte // The hash of the block.
-	PublicKey []byte // The public key used for signing.
-	Signature []byte // The signature of the block.
-	Timestamp int64  // The timestamp of the signing operation, in Unix seconds.
+	Msg       []byte // The message that the signature is signed over.
+	Signature []byte
+	Timestamp int64 // The timestamp of the signing operation, in Unix seconds.
 }
 
 func (s *SigningRecord) FromProto(sr *proto.SigningRecord) {
-	s.PublicKey = sr.PublicKey
-	s.BlockHash = sr.BlockHash
+	s.Msg = sr.Msg
 	s.Timestamp = sr.Timestamp
-	s.Signature = sr.Signature
+	s.Signature = sr.EotsSig
+}
+
+func getSignRecordKey(pk, chainID []byte, height uint64) []byte {
+	// calculate the randomn hash of the key concatenated with chainID and height
+	digest := hmac.New(sha256.New, pk)
```

Comment:
hmm I wonder whether just concatenating bytes wouldn't be be better as key.

If the key is `fpPk || height || chainID` then finding all records for `fp` is just finding all keys with prefix `fpPk || ..`. I am not sure what this hashing gives us except decreasing key slightly. 

(this is not blocking though, we can disscuss it in separate issue)

Reply:
I agree. Will do it in this PR as db schema change in the future is troublesome

Reply:
Done in [48ac5ee](https://github.com/babylonlabs-io/finality-provider/pull/199/commits/48ac5eea80c7c9ff9f994efa830c09aa1059b999)